tools/unit_testing/run_tests.py

```diff
@@ -277,8 +278,20 @@ def _main():
     # Create the environment, and the Macros.cmake file
     #
     #
-    configure(machobj, build_dir, ["CMake"], compiler, mpilib, debug, os_)
-    machspecific = EnvMachSpecific(build_dir)
+    configure(machobj, build_dir, ["CMake"], compiler, mpilib, debug, os_,
+              unit_testing=True)
+    machspecific = EnvMachSpecific(build_dir, unit_testing=True)
+
+    machspecific.load_env(compiler, debug, mpilib)
+    os.environ["OS"] = os_
+    os.environ["COMPILER"] = compiler
+    os.environ["DEBUG"] = stringify_bool(debug)
+    os.environ["MPILIB"] = mpilib
+    if use_openmp:
+        os.environ["compile_threaded"] = "true"
+    os.environ["CC"] = find_executable("mpicc")
+    os.environ["FC"] = find_executable("mpif90")
+    os.environ["UNIT_TEST_HOST"] = socket.gethostname()
```

Comment:
This environment stuff is mostly just moved up from below. But note that I also now set UNIT_TEST_HOST.